zettarepl/dataset/data.py

```diff
@@ -78,13 +94,61 @@ def inspect_data(shell: Shell, dataset: str):
                 index = shell.ls(dst_properties["mountpoint"])
             except Exception as e:
                 logger.warning(
-                    "An exception occurred while listing dataset %r mountpoint %r: %r. Assuming dataset is not mounted",
-                    dataset, dst_properties["mountpoint"], e,
+                    "An exception occurred while listing dataset %r mountpoint %r on shell %r: %r. "
+                    "Assuming dataset is not mounted",
+                    dataset, dst_properties["mountpoint"], shell, e,
                 )
             else:
                 if dst_properties["snapdir"] == "visible" and ".zfs" in index:
                     index.remove(".zfs")
 
+                for excluded in exclude:
+                    if excluded not in index:
+                        continue
+
+                    child_mountpoint = os.path.join(dst_properties["mountpoint"], excluded)
```

Comment:
Do we handle legacy mountpoints?

Reply:
Yes, if dataset is mounted using legacy mountpoint, we don't read its contents and use fallback method (`used > 10MB`) to determine whether it has data.